### PR TITLE
Solve numerical stability problem  (#91)

### LINKED ISSUE #91: Numerical stability during cat state breeding
In the proess of cat state breeding, if cat state amplitude >4.7, `measure_homodyne`  can't work, in that some entries of c_tilde after exponential become inf. 

Image

If mask the inf to a fix large value, `measure_homodyne` works fine, while the weights of `state_measured` turn into nan.

Image

Image

### PR DESCRIPTION
**Pull request by Hugh-888**
Fix #91

### CODE REVIEW

File: src/deepquantum/photonic/qmath.py [hunk 1]
```diff
@@ -346,10 +346,10 @@ def sample_homodyne_fock(
     nmode: int,
     cutoff: int,
     shots: int = 1,
-    den_mat: bool = False
+    den_mat: bool = False,
+    x_range: float = 15
```
Comment (sansiro77) [inline src/deepquantum/photonic/qmath.py]: nbin是不是也可以放到参数里？

File: src/deepquantum/photonic/state.py [hunk 2]
```diff
@@ -366,8 +366,8 @@ def marginal(self, wire: int, qvec: torch.Tensor, phi: float = 0., plot: bool =
         mean = mean[..., 0, 0].unsqueeze(1)
         prefactor = 1 / (torch.sqrt(2 * torch.pi * cov)) # (batch, 1, ncomb)
         # (batch, npoints, ncomb)
-        marginal_vals = self.weight.unsqueeze(1) * prefactor * torch.exp(-0.5 * (qvec.reshape(-1, 1) - mean)**2 / cov)
-        marginal_vals = marginal_vals.sum(2).real # (batch, npoints)
+        log_marginal_vals = torch.log(self.weight.unsqueeze(1)) + torch.log(prefactor) - 0.5 * (qvec.reshape(-1, 1) - mean)**2 / cov
```
Comment (sansiro77) [inline src/deepquantum/photonic/state.py]: 这行有点长了。用`torch.log(self.weight.unsqueeze(1) *prefactor)`的效果有什么差别吗？`log_marginal_vals `也可以换个更短的变量名

**Review by sansiro77 — APPROVED** [review]
GJ
